kpops/utils/yaml_loading.py

```diff
@@ -37,3 +39,54 @@ def substitute(input: str, substitution: Mapping[str, Any] | None = None) -> str
     if not substitution:
         return input
     return Template(input).safe_substitute(**substitution)
+
+
+def substitute_nested(input: str, max_repetitions: int = 500, **kwargs) -> str:
+    """Allow for multiple substitutions to be passed.
+
+    Will make as many passes as needed to substitute all possible placeholders.
+    A ceiling is set to avoid infinite loops.
+
+    HINT: If :param input: is a ``Mapping`` that you converted into ``str``,
+    You can pass it as a string, and as a ``Mapping`` to enable self-reference.
+
+    :Example:
+
+    >>> substitution = {
+            "a": "0",
+            "b": "${a}",
+            "c": "${b}",
+            "d": "${a}",
+        }
+    >>> input = "${a}, ${b}, ${c}, ${d}"
+    >>> print("Substituted string: " + substitute_nested(input, **substitution))
+    0, 0, 0, 0
+
+    :param input: The raw input containing $-placeholders
+    :type input: str
+    :param max_repetitions: The max number of consecutive substitutions to be carried out, defaults to 500
+    :type max_repetitions: int, optional
+    :param **kwargs: Substitutions
+    :raises Exception: Substitution was repeated {counter} times and placeholders still exist, check for loops
+    :return: Substituted input string
+    :rtype: str
+    """
+    if not {**kwargs}:
```

Comment:
I believe `{**kwargs}` is unnecessary, just use `kwargs` directly

Reply:
indeed, will fix it

Reply:
done